deall.ii dev build will not compile with gcc version 4.4.7

```
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc: In member function ‘void dealii::FESystem<dim, spacedim>::compute_fill(const dealii::Mapping<dim, spacedim>&, const typename dealii::Triangulation<dim, spacedim>::cell_iterator&, unsigned int, unsigned int, const dealii::Quadrature<dim_1>&, dealii::CellSimilarity::Similarity, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const dealii::internal::FEValues::MappingRelatedData<dim, spacedim>&, dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim>&) const [with int dim_1 = 1, int dim = 1, int spacedim = 1]’:
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc:1046:   instantiated from ‘void dealii::FESystem<dim, spacedim>::fill_fe_values(const dealii::Mapping<dim, spacedim>&, const typename dealii::Triangulation<dim, spacedim>::cell_iterator&, const dealii::Quadrature<dim>&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const dealii::internal::FEValues::MappingRelatedData<dim, spacedim>&, dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim>&, dealii::CellSimilarity::Similarity) const [with int dim = 1, int spacedim = 1]’
/gpfs/home/jvs5482/lib/devDeal/build/source/fe/fe_system.inst:2:   instantiated from here
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc:1305: error: no matching function for call to ‘bind(<unresolved overloaded function type>, const dealii::FESystem<1, 1>* const, const boost::reference_wrapper<const dealii::Mapping<1, 1> >, const boost::reference_wrapper<const dealii::TriaIterator<dealii::CellAccessor<1, 1> > >, std::pair<unsigned int, unsigned int>, const boost::reference_wrapper<const dealii::Quadrature<1> >, dealii::CellSimilarity::Similarity, std::pair<const dealii::Mapping<1, 1>::InternalDataBase*, const dealii::Mapping<1, 1>::InternalDataBase*>, unsigned int&, const boost::reference_wrapper<const dealii::internal::FEValues::MappingRelatedData<1, 1> >, const boost::reference_wrapper<dealii::internal::FEValues::FiniteElementRelatedData<1, 1> >)’
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc: In member function ‘void dealii::FESystem<dim, spacedim>::compute_fill(const dealii::Mapping<dim, spacedim>&, const typename dealii::Triangulation<dim, spacedim>::cell_iterator&, unsigned int, unsigned int, const dealii::Quadrature<dim_1>&, dealii::CellSimilarity::Similarity, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const dealii::internal::FEValues::MappingRelatedData<dim, spacedim>&, dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim>&) const [with int dim_1 = 0, int dim = 1, int spacedim = 1]’:
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc:1065:   instantiated from ‘void dealii::FESystem<dim, spacedim>::fill_fe_face_values(const dealii::Mapping<dim, spacedim>&, const typename dealii::Triangulation<dim, spacedim>::cell_iterator&, unsigned int, const dealii::Quadrature<(dim - 1)>&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const dealii::internal::FEValues::MappingRelatedData<dim, spacedim>&, dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim>&) const [with int dim = 1, int spacedim = 1]’
/gpfs/home/jvs5482/lib/devDeal/build/source/fe/fe_system.inst:2:   instantiated from here
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc:1305: error: no matching function for call to ‘bind(<unresolved overloaded function type>, const dealii::FESystem<1, 1>* const, const boost::reference_wrapper<const dealii::Mapping<1, 1> >, const boost::reference_wrapper<const dealii::TriaIterator<dealii::CellAccessor<1, 1> > >, std::pair<unsigned int, unsigned int>, const boost::reference_wrapper<const dealii::Quadrature<0> >, dealii::CellSimilarity::Similarity, std::pair<const dealii::Mapping<1, 1>::InternalDataBase*, const dealii::Mapping<1, 1>::InternalDataBase*>, unsigned int&, const boost::reference_wrapper<const dealii::internal::FEValues::MappingRelatedData<1, 1> >, const boost::reference_wrapper<dealii::internal::FEValues::FiniteElementRelatedData<1, 1> >)’
/gpfs/home/jvs5482/lib/devDeal/dealii/source/fe/fe_system.cc: In member function ‘void dealii::FESystem<dim, spacedim>::compute_fill(const dealii::Mapping<dim, spacedim>&, const typename dealii::Triangulation<dim, spacedim>::cell_iterator&, unsigned int, unsigned int, const dealii::Quadrature<dim_1>&, dealii::CellSimilarity::Similarity, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const typename dealii::Mapping<dim, spacedim>::InternalDataBase&, const dealii::internal::FEValues::MappingRelatedData<dim, spacedim>&, dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim>&) const [with int dim_1 = 1, int dim = 1, int spacedim = 2]’:
```

--- a/include/deal.II/fe/fe_system.h
+++ b/include/deal.II/fe/fe_system.h
@@ -1036,12 +1036,12 @@ private:
   template <int dim_1>
   void
   compute_fill_one_base (const Mapping<dim,spacedim>                      &mapping,
-                         const typename Triangulation<dim,spacedim>::cell_iterator &cell,
+                         const std::pair<typename Triangulation<dim,spacedim>::cell_iterator,
+                         CellSimilarity::Similarity>       cell_and_similarity,
                          const std::pair<unsigned int,unsigned int>        face_sub_no,
                          const Quadrature<dim_1>                          &quadrature,
-                         const CellSimilarity::Similarity                  cell_similarity,
                          const std::pair<const typename Mapping<dim,spacedim>::InternalDataBase *,
-                         const typename Mapping<dim,spacedim>::InternalDataBase *> mapping_internal,
+                         const typename Mapping<dim,spacedim>::InternalDataBase *> mapping_and_fe_internal,
                          const unsigned int                                base_element,
                          const internal::FEValues::MappingRelatedData<dim,spacedim> &mapping_data,
                          internal::FEValues::FiniteElementRelatedData<dim,spacedim> &output_data) const;

--- a/source/fe/fe_system.cc
+++ b/source/fe/fe_system.cc
@@ -1095,17 +1095,20 @@ template <int dim_1>
 void
 FESystem<dim,spacedim>::
 compute_fill_one_base (const Mapping<dim,spacedim>                      &mapping,
-                       const typename Triangulation<dim,spacedim>::cell_iterator &cell,
+                       const std::pair<typename Triangulation<dim,spacedim>::cell_iterator,
+                       CellSimilarity::Similarity>       cell_and_similarity,
                        const std::pair<unsigned int, unsigned int>       face_sub_no,
                        const Quadrature<dim_1>                          &quadrature,
-                       const CellSimilarity::Similarity                  cell_similarity,
                        const std::pair<const typename Mapping<dim,spacedim>::InternalDataBase *,
-                       const typename Mapping<dim,spacedim>::InternalDataBase *> mapping_internal,
+                       const typename Mapping<dim,spacedim>::InternalDataBase *> mapping_and_fe_internal,
                        const unsigned int                                base_no,
                        const internal::FEValues::MappingRelatedData<dim,spacedim> &mapping_data,
                        internal::FEValues::FiniteElementRelatedData<dim,spacedim> &output_data) const
 {
-  const InternalData &fe_data = static_cast<const InternalData &> (*mapping_internal.second);
+  const typename Triangulation<dim,spacedim>::cell_iterator cell            = cell_and_similarity.first;
+  const CellSimilarity::Similarity                          cell_similarity = cell_and_similarity.second;
+
+  const InternalData &fe_data = static_cast<const InternalData &> (*mapping_and_fe_internal.second);
   const unsigned int n_q_points = quadrature.size();
 
   const FiniteElement<dim,spacedim> &
@@ -1164,14 +1167,14 @@ compute_fill_one_base (const Mapping<dim,spacedim>                      &mapping
   // copied from base_data to data on each face, therefore use
   // base_fe_data.update_flags.
   if (face_sub_no.first==invalid_face_number)
-    base_fe.fill_fe_values(mapping, cell, *cell_quadrature, *mapping_internal.first,
+    base_fe.fill_fe_values(mapping, cell, *cell_quadrature, *mapping_and_fe_internal.first,
                            base_fe_data, base_data, base_data, cell_similarity);
   else if (face_sub_no.second==invalid_face_number)
     base_fe.fill_fe_face_values(mapping, cell, face_sub_no.first,
-                                *face_quadrature, *mapping_internal.first, base_fe_data, base_data, base_data);
+                                *face_quadrature, *mapping_and_fe_internal.first, base_fe_data, base_data, base_data);
   else
     base_fe.fill_fe_subface_values(mapping, cell, face_sub_no.first, face_sub_no.second,
-                                   *face_quadrature, *mapping_internal.first, base_fe_data, base_data, base_data);
+                                   *face_quadrature, *mapping_and_fe_internal.first, base_fe_data, base_data, base_data);
 
   // now data has been generated, so copy it. we used to work by
   // looping over all base elements (i.e. this outer loop), then over
@@ -1306,10 +1309,9 @@ compute_fill (const Mapping<dim,spacedim>                      &mapping,
         += Threads::new_task (std_cxx11::function<void ()>(std_cxx11::bind (&FESystem<dim,spacedim>::template compute_fill_one_base<dim_1>,
                                                            this,
                                                            std_cxx11::cref(mapping),
-                                                           std_cxx11::cref(cell),
+                                                           std::make_pair(cell, CellSimilarity::none),
                                                            std::make_pair(face_no, sub_no),
                                                            std_cxx11::cref(quadrature),
-                                                           CellSimilarity::none,
                                                            std::make_pair(&mapping_internal,
                                                                &fedata),
                                                            base_no,

